test/presubmit-tests.sh

```diff
@@ -26,20 +26,41 @@ source $(dirname $0)/../vendor/github.com/tektoncd/plumbing/scripts/presubmit-te
 function run() {
     folder=$1
     header "${folder}"
-    shift
-    pushd $(dirname $0)/../${folder} || return 1
-    if [[ -f ./test/presubmit-tests.sh ]]; then
-        ./test/presubmit-tests.sh $@ || exited=1
-    else
-        echo "Skip due to no `./test/presubmit-tests.sh` file"
+    if should_test_folder $folder ; then
+      shift
+      pushd $(dirname $0)/../${folder} || return 1
+      if [[ -f ./test/presubmit-tests.sh ]]; then
+          ./test/presubmit-tests.sh $@ || exited=1
+      else
+          echo "Skip due to no `./test/presubmit-tests.sh` file"
+      fi
+      popd >/dev/null
+      return $exited
     fi
-    popd >/dev/null
-    return $exited
+    echo "Skip - no files changed"
+    return 0
 }
 
-run webhooks-extension $@ || exit 1
-run catalogs $@ || exit 1
-run octant-plugin $@ || exit 1
-run tekdoc  $@ || exit 1
-run results $@ || exit 1
-run generators $@ || exit 1
+function should_test_folder() {
+    # If initialize_environment failed to identify the changed files, fall back to testing everything.
+    if [[ -z "$(cat ${CHANGED_FILES})" ]]; then
+        echo "Cannot determine changed files.  Testing all projects." && return 0
+    fi
+    for file in $(cat "${CHANGED_FILES}"); do
+        # If changed file is in the folder then test that folder.
+        echo $file | grep -q "^$1/.*" && echo "$file is modified so testing project $1" && return 0
+        # If changed file is outside all project folders then test every folder.
+        echo $file | grep -q -v $inanyprojectregex && echo "$file is modified so testing all projects" && return 0
+    done
+    return 1
+}
+
+# Get list of changed files
+initialize_environment
+
+projects="webhooks-extension catalogs octant-plugin tekdoc results generators task-loops"
```

Comment:
I think the complete list here should be `catalogs commit-status-tracker generators helm hub oci octant-plugin pipeline results task-loops tekdoc tekton-listener webhooks-extension`

Reply:
I used the list of projects in the existing presubmit-tests script.  The other subprojects not in the list don't have a presubmit-tests script so nothing would happen for them.

Reply:
Oh wait, I see.  Even though those projects don't have tests, they need to be in the list so that changes to those projects aren't considered "global" which would cause every project's tests to run.  I've updated the list.